Add String method for functions

stevelinton opened the pull request:
A bit of a hack as it uses string streams rather than modifying the kernel

fingolfin commented on lib/function.gi:
```diff
@@ -59,3 +59,16 @@ function(func)
 end);
+
+
+InstallMethod(String, "for a function, using string stream", [IsFunction] , 
```
remove space before comma?

fingolfin commented on lib/function.gi:
```diff
@@ -59,3 +59,16 @@ function(func)
+        function(fun)
+    local  s, str;
+    s := "";
+    str := OutputTextString(s,true);
```
insert space after comma?

fingolfin commented on lib/function.gi:
```diff
@@ -59,3 +59,16 @@ function(func)
+
+
+InstallMethod(String, "for a function, using string stream", [IsFunction] , 
+        function(fun)
```
Why this odd indentation?

stevelinton commented:
@fingolfin I've addressed your comments. I've also made the original method a method for DisplayString. The String method now returns a version with  normalised whitespace, which is
ugly for large functions, but I think more appropriate for String.

fingolfin commented on lib/function.gi:
```diff
@@ -59,3 +59,26 @@ function(func)
+    str := OutputTextString(s, true);
+    PrintTo(str, fun);
+    CloseStream(str);
+    Add(s,'\n');    
```
Space after comma?

fingolfin commented on lib/function.gi:
```diff
@@ -59,3 +59,26 @@ function(func)
+    s := "";
+    str := OutputTextString(s, true);
+    PrintTo(str, fun);
+    CloseStream(str);
```
Super-nitpick, I am not expecting any action on it, but needed to get it off my chest: When I see "str" I think "string", but here it is "stream", while "s" is a string... This function is short, and so it's harmless, but I'd still rename at least one of them (probably `str` to `stream`, as I see strings more more often than streams, at least in GAP code)

stevelinton commented:
Fixed nitpicks. Just wait for the tests to make sure I didn't break something.
